bindings/gumjs/gumcmodule.c

```diff
@@ -173,33 +232,37 @@ gum_cmodule_free (GumCModule * cmodule)
     gum_memory_free (GSIZE_TO_POINTER (r->base_address), r->size);
   }
 
-  gum_cmodule_drop_metadata (cmodule);
+  gum_cmodule_drop_metadata (GUM_CMODULE (_cmodule));
 
-  g_slice_free (GumCModule, cmodule);
+  G_OBJECT_CLASS (gum_tcc_cmodule_parent_class)->dispose (_cmodule);
 }
 
-const GumMemoryRange *
-gum_cmodule_get_range (GumCModule * self)
+static const GumMemoryRange *
+gum_tcc_cmodule_get_range (GumCModule * _self)
 {
+  GumTccCModule * self = (GumTccCModule *) _self;
```

Comment:
```suggestion
gum_tcc_cmodule_get_range (GumCModule * cm)
{
  GumTccCModule * self = GUM_TCC_CMODULE (cm);
```

This is the convention I've been using so far. Also good to use cast macros so that type checks can be performed when built with `-Db_ndebug=false` in `config.mk`.